nima_core/llm_client.py

```
@@ -1,302 +1,175 @@
+                messages = []
+                if system:
+                    messages.append({"role": "system", "content": system})
+                messages.append({"role": "user", "content": prompt})
+                if config["provider"] == "anthropic":
+                    payload = {"model": config["model"], "max_tokens": max_tokens, "messages": [m for m in messages if m["role"] != "system"]}
+                    if system:
+                        payload["system"] = system
+                else:
+                    payload = {"model": config["model"], "max_tokens": max_tokens, "messages": messages}
+                data = json.dumps(payload).encode("utf-8")
+                req = urllib.request.Request(config["endpoint"], data=data, headers=config["headers"])
```

Ruff (0.15.6)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)